### PR TITLE
chore: remove `eslint-config-kentcdodds`

### PR DESCRIPTION
**Pull request by lesha1201**
## Checks

- [x] I have read the [contributing guidelines](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/CONTRIBUTING.md).

## Context

These changes were initially introduced in https://github.com/testing-library/eslint-plugin-testing-library/pull/852

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -54,6 +66,24 @@ module.exports = {
...
+				// TypeScript and @typescript-eslint.
+				'import/default': 'off',
+				'import/export': 'off',
+				'import/namespace': 'off',
+				'import/no-unresolved': 'off',
+			},
+			settings: {
+				'import/resolver': {
+					node: {
+						extensions: ['.js', '.jsx', '.ts', '.tsx'],
```
Comment (Belco90) [inline .eslintrc.js:81]: We don't need jsx and tsx in this repo.

**Review by Belco90 — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (lesha1201) [reply .eslintrc.js]: Done

**Comment by lesha1201** [comment]
@Belco90 By the way, do you have some conventions/rules in mind that you want to enforce? I would like to improve ESLint setup  later but not very familiar with the codebase yet and its conventions.

**Comment by lesha1201** [comment]
@Belco90 CI will fail for Node v12 because of this https://github.com/testing-library/eslint-plugin-testing-library/pull/857#issuecomment-1855338100. Previously, `eslint-config-kentcdodds` were taking precedence over our `extends: ['plugin:jest/recommended']` and because of that ESLint were using an old version of `eslint-plugin-jest` which support Node.js v12. But now it uses our version of `eslint-plugin-jest` which doesn't support Node.js v12.

image

**Comment by Belco90** [comment]
> @Belco90 By the way, do you have some conventions/rules in mind that you want to enforce? I would like to improve ESLint setup later but not very familiar with the codebase yet and its conventions.

I have something in mind, but let's do that in a follow-up PR.

**Review by Belco90 — APPROVED** [review]
Sorry for the late response! The Christmas break was way busier than I thought, so I'm slowly coming back to this now.

**Comment by Belco90** [comment]
@all-contributors please add @lesha1201 for maintenance

**Pull request merged**